### ISSUE TITLE
Discuss DX when jumping to definitions

### ISSUE THREAD

**Issue by sodic** [wasp-lang/wasp#1847]
Most of Wasp's constructs are users implementation wrapped in some kind of machinery which the user can then call to access their original functionality.

The central question is: How do we ideally want to handle jump-to-definition for such constructs (e.g,. Queries, Actions, ...).

We can either:
  1. **Reflect the runtime** - This is what Wasp currently does (as of 0.12.0). The jump represents what truly happens.
     - Jump-to-definition takes the developer to the Wasp-generated machinery that really gets called in runtime.
     - **Example:** Jumping to the definition of a Query on the frontend takes the developer to a frontend function that this makes the API call the backend.
2. **Reflect the intuition** - This is what's popular these days (e.g., tRPC). The jump sacrifices the truth for the sake of what the developer (hopefully) cares about - seeing their own implementation.
   - Jump-to-definition takes the developer to the original functionality they implemented, pretending that all the machinery doesn't exist
   - **Example:** Jumping to the definition of a Query on the frontend takes the their definition of a Query imported into a Wasp file.

**Comment by cprecioso** [comment]
@sodic You already did some work to allow jump to definition. Is this issue up-to-date?

**Comment by sodic** [comment]
Yes, very much so - it was created after the restructuring in 0.12.0.

The core question is "Do we want to lie when jumping to definition and take users to what they probably want to see?"
More technical info in the description.
